src/ts/core/features/filter-search.ts

```diff
@@ -0,0 +1,130 @@
+import {Feature, Settings} from '../settings'
+import {browser} from 'webextension-polyfill-ts'
+
+export const config: Feature = { // An object that describes new feature we introduce
+    id: 'filter_search',  // Feature id - any unique string would do
+    name: 'Filter Search',  // Feature name - would be displayed in the settings menu
+    //warning: 'Experimental feature',
+    enabledByDefault: true,
+}
+
+const checkSettingsAndSetupFilterSearchToggle = () => {
+    Settings.isActive('filter_search').then(active => {
+        toggleFilterSearch(active)
+    })
+}
+checkSettingsAndSetupFilterSearchToggle()
+
+browser.runtime.onMessage.addListener(async message => {
+    if (message === 'settings-updated') {
+        checkSettingsAndSetupFilterSearchToggle()
+    }
+})
+
+const toggleFilterSearch = (active: boolean) => {
+    const jsDoc = document;
```

Comment:
I don't think extracting this into a variable adds much

Reply:
Do you mean the “const toggleFilterSearch...” part? I just followed the Live Preview feature because doing similar thing so re-used that code. Is this ok?

Reply:
I meant the `jsDoc` one. general structure seems alright :)